web/website/content/posts/2023-05-07-whats-next-for-prql.md

```diff
@@ -0,0 +1,118 @@
+  devoid of commercial interests; this avoids conflicts of interest, especially
+  crucial at the language layer.
+
+## Weaknesses
+
+In summary, we have one major shortcoming: we haven't yet established a flywheel
+consisting of:
+
+- Users — using the language, providing feedback on what works and what doesn't.
+  They then become evangelists for more users, attracting tools that want to
+  integrate PRQL. (Some technical users may even become contributors.)
+- Tools — incorporating PRQL into their offerings, broadening PRQL's audience,
+  and giving us valuable insights into which language features succeed or fail.
+- Developers — building PRQL! Due to the Users & Tools, this feels like an
+  impactful and exciting project rather than a personal hobby (although that's
+  fine too!).
+
+Specifically, on each of those:
+
+- Very few actual users (compared to stars)
+- No outstanding implementation in any data tool. While there are decent
+  bindings, it's unclear where to direct someone who simply wants to write PRQL
+  and view data. Some attempts (e.g., dbt-prql) faltered on technical grounds.
+- Though the Developers aspect of the three-legged stool has consistently been
+  the strongest, even here we've seen a decline in engagement:
+  - The past few months have had fluctuating levels of contributions outside the
+    core team — with a surge in Q1, but slowing since then.
+  - Recently, even the core team's pace has decreased (although there's always
+    variance, and I'm not suggesting a quick fix by us working harder).
+
+## Finding the Flywheel
+
+The most crucial step is enabling people to use PRQL. Much will follow — we'll
+know which features and bugs to tackle, be able to showcase satisfied users, and
+so on.
+
+Some options:
+
+- Revisit `dbt`. I had envisioned this as a particularly fruitful integration —
+  for both PRQL and dbt.
+
+  - The monkeypatch plugin couldn't technically achieve everything needed.
+  - I also failed to have the dbt-core PR merged — it received a brief review
+    before being dropped.
+  - The dbt team remains interested in merging an improved version of the
+    dbt-core plugin (but won't guarantee a merge, which I can understand, though
+    makes it a less compelling investment)
+  - I'm hesitant to invest more time given this track record but open to it if
+    others believe it's a worthwhile bet.
+
+- Enhance language-specific integrations
+
+  - The Jupyter integration is "OK". Python is very popular so if it were
+    possible to make it excellent, then it could be great. But can it be great?
+    Can we get good editor features in a notebook? I _think_ we can get syntax
+    highlighting.
+    - TODO: how much integration can we actually do in Jupyter? Could we have an
+      LSP?
+  - The R extension looks great. I don't have much experience with R and haven't
+    explored it extensively. Do others have opinions? Is the R audience enough
+    to make a project-wide bet on R?
```

Comment:
In my personal opinion, the most valuable aspect of prqlr is the ability to easily embed PRQL code blocks in Quarto documents.
This also can be done with IPython magic with pyprql, but it is interpreted as a Python code block and does not look very good. (quarto-dev/quarto-cli#4227)

Embedding PRQL code blocks into Quarto documents works very well, as in https://github.com/eitsupi/querying-with-prql, but the problem is that there is no way to run PRQL code interactively.
SQL code blocks can be executed interactively in the RStudio IDE, but if you want PRQL code blocks to work the same way, we will need to add functionality to the RStudio IDE. (Related to rstudio/rstudio#12767)
Also, not even SQL code blocks are currently executable in VSCode.
I am thinking it would be more realistic to wait for Quarto to be able to shape IPython Magic's embedded code blocks into pretty code blocks rather than making PRQL code blocks interactively executable in RStudio or VSCode.